src/Components/Form/Form.test.js

```diff
@@ -0,0 +1,170 @@
+    const mockActivities = [];
+    const mockSetActivities = jest.fn();
+    const mockUpdateActivityAnswers = jest.fn();
+    const mockSetHistory = jest.fn();
+    const mockGoBack = jest.fn();
+    const mockResetState = jest.fn();
+    
+    render(
+      <MemoryRouter>
+        <Form 
+         activities={mockActivities}
+         setActivities={mockSetActivities}
+         updateActivityAnswers={mockUpdateActivityAnswers}
+         setHistory={mockSetHistory}
+         goBack={mockGoBack}
+         resetState={mockResetState}
+        />
+      </MemoryRouter>
+    )
+
+    await waitFor(() => screen.getByText('Which activities excite you right now?'))
+    userEvent.click(screen.getByText('music'));
+    userEvent.click(screen.getByText('card games'));
+    userEvent.click(screen.getByRole('button', {name: 'next'}));
+    const nextQuestion = await waitFor(() => screen.getByText('Which music genre(s)?'))
+    expect(nextQuestion).toBeInTheDocument();
+    expect(mockSetActivities).toHaveBeenCalledTimes(1);
+    userEvent.click(screen.getByText('Country'));
+    userEvent.click(screen.getByRole('button', {name: 'next'}))
+    expect(screen.getByText('Please select at least one option for each question')).toBeInTheDocument();
+    userEvent.click(screen.getByText('2000\'s'));
+    userEvent.click(screen.getByRole('button', {name: 'next'}));
+    const gameQuestion = await waitFor(() => screen.getByText('How many people are playing?'));
+    expect(gameQuestion).toBeInTheDocument();
+  })
+
+  it('should show the user the previous set of questions if back button is clicked', async () => {
+    const mockActivities = [];
+    const mockSetActivities = jest.fn();
+    const mockUpdateActivityAnswers = jest.fn();
+    const mockSetHistory = jest.fn();
+    const mockGoBack = jest.fn();
+    const mockResetState = jest.fn();
+    
+    render(
+      <MemoryRouter>
+        <Form 
+         activities={mockActivities}
+         setActivities={mockSetActivities}
+         updateActivityAnswers={mockUpdateActivityAnswers}
+         setHistory={mockSetHistory}
+         goBack={mockGoBack}
+         resetState={mockResetState}
+        />
+      </MemoryRouter>
+    )
+
+    await waitFor(() => screen.getByText('Which activities excite you right now?'))
+    userEvent.click(screen.getByText('music'));
+    userEvent.click(screen.getByText('card games'));
+    userEvent.click(screen.getByRole('button', {name: 'next'}));
+    const nextQuestion = await waitFor(() => screen.getByText('Which music genre(s)?'))
+    expect(nextQuestion).toBeInTheDocument();
+    expect(mockSetHistory).toHaveBeenCalledTimes(1);
+    userEvent.click(screen.getByRole('button', {name: 'back'}));
+    const prevQuestion = await waitFor(() => screen.getByText('Which activities excite you right now?'));
+    expect(prevQuestion).toBeInTheDocument();
+    expect(mockGoBack).toHaveBeenCalledTimes(1);
```

Comment:
Good call integrating testing for the back button, glad to see it wasn't too much of a pain to do.

Reply:
Of course!